### PR TITLE
Small enhancements

### PR DESCRIPTION
**Pull request by vtelensky**
- Moving method for returning index HTML into @Controller component 
- Changing to BoxController to BoxRestController and also anotating as @RestController component
- Splitting one method for handeling  all exceptions in GlobalControllerExceptionHandler to separated method for specific exceptions
- Moving classes under NotFoundException to exception package
- GlobalControllerExceptionHandler under util package

### CODE REVIEW

File: src/main/java/cz/sparko/boxitory/service/BoxRepository.java [hunk 1]
```diff
@@ -37,7 +38,7 @@
      * @param boxProvider provider of box to find
      * @param boxVersion  version of box to find
      * @return {@link BoxStream} found by given parameters, {@link Optional#empty} when box file not found
-     * @throws cz.sparko.boxitory.conf.NotFoundException when box does not exist or don't have given provider or dont
+     * @throws NotFoundException when box does not exist or don't have given provider or dont
```
Comment (sparkoo) [inline src/main/java/cz/sparko/boxitory/service/BoxRepository.java]: please, make this `@link`, same as in `latestVersionOfBox` javadoc

File: src/main/java/cz/sparko/boxitory/controller/util/GlobalControllerExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package cz.sparko.boxitory.controller.util;
+
+import cz.sparko.boxitory.exception.NotFoundException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class GlobalControllerExceptionHandler {
```
Comment (sparkoo) [inline src/main/java/cz/sparko/boxitory/controller/util/GlobalControllerExceptionHandler.java:10]: very nice, thanks!

**Review by sparkoo — COMMENTED** [review]
thank you, nice improvements. Just please fix one trivial in javadoc and I'll merge it.

File: src/main/java/cz/sparko/boxitory/controller/util/GlobalControllerExceptionHandler.java (hunk 2 above)
Comment (vtelensky) [reply src/main/java/cz/sparko/boxitory/controller/util/GlobalControllerExceptionHandler.java:10]: No problem, thanks.

File: src/main/java/cz/sparko/boxitory/service/BoxRepository.java (hunk 1 above)
Comment (vtelensky) [reply src/main/java/cz/sparko/boxitory/service/BoxRepository.java]: Done

**Pull request merged**